kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/steps/KafkaSteps.java

```diff
@@ -65,39 +65,35 @@ public static void createTopic(String deployNamespace, String topicName, String
-        kubeClient().getClient().run().inNamespace(deployNamespace).withNewRunConfig()
-                .withImage(Constants.TEST_CLIENTS_IMAGE)
-                .withName(podName)
-                .withRestartPolicy(NEVER_POLICY)
-                .withCommand(ADMIN_CLIENT_COMMAND)
-                .withArgs(TOPIC_COMMAND, "delete", BOOTSTRAP_ARG + bootstrap, "--topic=" + topicName)
-                .done();
+        LOGGER.atDebug().setMessage("Deleting '{}' topic").addArgument(topicName).log();
+        String name = Constants.KAFKA_ADMIN_CLIENT_LABEL + "-delete";
+        List<String> args = Arrays.asList(TOPIC_COMMAND, "delete", BOOTSTRAP_ARG + bootstrap, "--topic=" + topicName);
+
+        Job adminClientJob = TestClientsJobTemplates.defaultAdminClientJob(name, args).build();
+        kubeClient().getClient().batch().v1().jobs().inNamespace(deployNamespace).resource(adminClientJob).create();
 
+        String podName = KafkaUtils.getPodNameByLabel(deployNamespace, "app", name, Duration.ofSeconds(30));
         DeploymentUtils.waitForPodRunSucceeded(deployNamespace, podName, Duration.ofSeconds(30));
-        String log = kubeClient().logsInSpecificNamespace(deployNamespace, podName);
-        LOGGER.debug("Admin client delete pod log: {}", log);
+        LOGGER.atDebug().setMessage("Admin client delete pod log: {}").addArgument(kubeClient().logsInSpecificNamespace(deployNamespace, podName)).log();
     }
 
     private static boolean topicExists(String deployNamespace, String topicName, String bootstrap) {
-        LOGGER.info("Deleting topic {}", topicName);
-        String podName = Constants.KAFKA_ADMIN_CLIENT_LABEL + "-list";
-        kubeClient().getClient().run().inNamespace(deployNamespace).withNewRunConfig()
-                .withImage(Constants.TEST_CLIENTS_IMAGE)
-                .withName(podName)
-                .withRestartPolicy(NEVER_POLICY)
-                .withCommand(ADMIN_CLIENT_COMMAND)
-                .withArgs(TOPIC_COMMAND, "list", BOOTSTRAP_ARG + bootstrap)
-                .done();
+        LOGGER.debug("List '{}' topic", topicName);
+        String name = Constants.KAFKA_ADMIN_CLIENT_LABEL + "-list";
+        List<String> args = Arrays.asList(TOPIC_COMMAND, "list", BOOTSTRAP_ARG + bootstrap);
+
+        Job adminClientJob = TestClientsJobTemplates.defaultAdminClientJob(name, args).build();
+        kubeClient().getClient().batch().v1().jobs().inNamespace(deployNamespace).resource(adminClientJob).create();
 
+        String podName = KafkaUtils.getPodNameByLabel(deployNamespace, "app", name, Duration.ofSeconds(30));
         DeploymentUtils.waitForPodRunSucceeded(deployNamespace, podName, Duration.ofSeconds(30));
         String log = kubeClient().logsInSpecificNamespace(deployNamespace, podName);
-        LOGGER.debug("Admin client list pod log: {}", log);
-        return log.contains(topicName);
+        List<String> existingTopicNames = Arrays.stream(log.split("\n")).toList();
```

Comment:
Java 11 made this easier:

```suggestion
        List<String> existingTopicNames = log.lines().toList();
```

which leads you to:
```
return log.lines().anyMatch(topicName::equals)
```

but you lose the ability to log.  I'd keep the logging.